lib/EasyPost/Requestor.php

```diff
@@ -294,7 +294,22 @@ public function handleApiError($httpBody, $httpStatus, $response)
         if (!is_array($response) || !isset($response['error'])) {
             throw new Error("Invalid response object from API: HTTP Status: ({$httpStatus}) {$httpBody})", $httpStatus, $httpBody);
         }
-        throw new Error(is_array($response['error']) ? $response['error']['message'] : (!empty($response['error']) ? $response['error'] : ""), $httpStatus, $httpBody);
+
+        $message = "";
+
+        if (is_array($response['error'])) {
+            $message = $response['error']['message'];
+        } elseif (!empty($response['error'])) {
+            $message = $response['error'];
+        }
+
+        if (isset($response['errors']) && is_array($response['errors'])) {
+            $message = join(' ', $response['errors']);
+        } elseif (!empty($response['errors'])) {
+            $message = $response['errors'];
+        }
```

Comment:
@Justintime50 Looking again at this and trying to figure out why this is failing on the current implementation.

It appears that the only possibility is that **you are getting an array under `$response['error']['message']`**, at least in some, if not all, GLS error responses. Other possibilities would be handled correctly by the current implementation.

In that case, the below change to the current implementation would fix the issue and preserve current behavior in all cases that don't trigger the bug.

```suggestion
        if (is_array($response['error'])) {
            $message = is_array($response['error']['message'])
                ? json_encode($response['error']['message'])
                : $response['error']['message'];
        } elseif (!empty($response['error'])) {
            $message = $response['error'];
        }
```

Reply:
Interesting, can you share a complete error you've received from GLS? (the whole object), I believe you and I are talking about different kinds of errors (I'm referring to the more typical errors the API should be returning while I think you are referring to a GLS specific error that probably needs to be corrected at the API level instead of here as it may be putting that array of errors in the wrong field (which is then triggering this error). Seeing a complete example of a GLS error (and not the shipment create error I mentioned) would be helpful to determine that and next steps.

Reply:
Thanks for confirming this. This is actually a bug at the API level (most of our carrier integrations pass string errors in the `message` field while arrays of errors get passed in the `errors` field. If this carrier had been following this convention, this error would not be thrown as the PHP library here was built with the assumption that the API would return the right data. I'm speaking with our internal teams about this to see if we can get this resolved for GLS (GSO). As such, I'm not too keen on integrating this one-off fix for this carrier's improper error handling if we can instead fix this at the API level instead. I'll keep you posted on what I hear from them. As mentioned before, if a speedy fix is necessary for your integration, I'd suggest patching this locally until we can correct this at the API level or determine we must fix it here.